mava/utils/checkpointing.py

```diff
@@ -192,7 +193,16 @@ def restore_params(
         # Restore hidden states if required
         restored_hstates = None
         if restore_hstates and THiddenState is not None:
-            restored_hstates = THiddenState(**restored_learner_state_raw["hstates"])
+            if "hidden_state" in restored_learner_state_raw.keys():
```

Comment:
This is a very risk check for this, other hstates may also have "hidden_state" in the keys

Reply:
the other ones at the moment have hstate as a key this is why i don't think this is a problem

Reply:
another option is to just restore hstate as it is here but then inside the systems we add them to their types properly, what do you think about that?